src/message-stream.ts

```diff
@@ -43,6 +43,16 @@ const DEADLINE: status = 4;
  */
 const UNKNOWN: status = 2;
 
+/*!
+ * Max number of retries
+ */
+const DEFAULT_MAX_RETRIES = 3;
+
+/*!
+ * Time to reset retries counter.
+ */
+const RESET_RETRIES_COUNT_TIME_MS = 30000;
```

Comment:
Where does 30 seconds come from? We might want to get some input from @anguillanneuf here, but for some reason I seem to remember the PubSub team requesting that we retry for up to 5 minutes.

Reply:
@callmehiphop Idea is that if stream starts failing, almost all active/creating new stream will fail with in that timeout(30s by default) and it will retry for max number of retries specified(3 by default). 10-30s is good enough to determine if there is a connection/authentication issue to start with or if it needs refresh token/deadline exceeded after initial setup. 5 min to keep retrying is too long for this.

Reply:
I can understand the need to modify our retry mechanisms to support unauthenticated requests, but this change makes me a little nervous. My concern is that we start seeing an increase of errors that kill the subscriber that would otherwise be retried within that 5 minute duration. I would be interested to know how other clients handle this retrying this specific rpc. 

@sduskis any thoughts here?

Reply:
Adding @jadekler @jskeet for comments. I don't know if the 90 seconds found in GAPIC applies here.

Reply:
We definitely need to have a more holistic look at this topic. A couple of notes:

- I think an ideal state to be in is that the library creates no deadlines, instead opting to allow the user to specify a deadline. In some languages you can do this with contexts, but not all languages have contexts. We (client libs team, AC tools team) are investigating this topic later today.
- The list of codes to retry below seems too broad. I've noticed this in several pubsub clients. I've got an action item to chat with @sduskis about this. That said, immediate feedback on this PR is that I'm a thumbs down on this PR adding UNAUTHENTICATED to retry codes. It's a signal to the user that their config is borked and should not be retried.